### PR TITLE
fix: matcher function lost

### PR DESCRIPTION
**Pull request by closetool**
Signed-off-by: closetool <c299999999@qq.com>

Fix: #817

### CODE REVIEW

**Comment by nodece** [comment]
@closetool can you add a test for this change?

**Comment by closetool** [comment]
@nodece Sure

**Comment by closetool** [comment]
@nodece Added, plz review

**Comment by hsluoyz** [comment]
@nodece @closetool https://github.com/casbin/casbin/pull/819 is merged, is this PR still useful?

**Comment by closetool** [comment]
@hsluoyz Yeap, Model.ToText() still have a bug to fix.

File: model/model_test.go [hunk 1]
```diff
@@ -122,3 +123,32 @@ func TestModel_AddDef(t *testing.T) {
...
+		"r": "sub, obj, act",
+		"p": "sub, obj, act",
+		"e": "some(where (p.eft == allow))",
+		"m": "r.sub == p.sub && r.obj == p.obj && is_rw(r.act, p.act)",
+	}
+	expected := map[string]string{
+		"r": "sub, obj, act",
+		"p": "sub, obj, act",
+		"e": "some(where (p_eft == allow))",
+		"m": "r_sub == p_sub && r_obj == p_obj && is_rw(r_act, p_act)",
```
Comment (nodece) [inline model/model_test.go:139]: please add test `m = r_sub == p_sub && r_obj == p_obj && r_fuc(r_act, p_act) && testr_func(r_act, p_act)`.

**Review by nodece — CHANGES_REQUESTED** [review]

File: model/model_test.go (hunk 1 above)
Comment (closetool) [reply model/model_test.go]: fixed

Comment (nodece) [reply model/model_test.go]: @closetool don't remove the `r_func()` and `testr_func()`, which used to test `rPattern := regexp.MustCompile(`r_`)`, and you should also test the `regexp.MustCompile(`p_`)`.

Comment (nodece) [reply model/model_test.go]: @closetool Please update the test.

Comment (closetool) [reply model/model_test.go]: @nodece There's still some problems, we cann't figure out if it's a policy token or a user defined function for pattern has `p_` and `r_`, so `testr_func` will lead to error.

Comment (nodece) [reply model/model_test.go]: @closetool I found this problem, so I asked to use this master for test.

Comment (closetool) [reply model/model_test.go]: @nodece fixed

**Pull request merged**